Add target object word to "View on Etherscan" links

segun opened the pull request:
Fixes: #9476 

Explanation:  
We have "View on Etherscan" in a few places, and it means different things (accounts or tokens). We know this can be confusing to users. One simple way to mitigate this confusion is to update the wording in each case to clarify what is being viewed.

Manual testing steps:  
  - 
  - 
  -

segun committed:
Add target object word to "View on Etherscan" links
Fix for: #9476

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

brad-decker commented on app/_locales/en/messages.json:
```diff
@@ -2754,13 +2754,13 @@
   },
   "viewOnCustomBlockExplorer": {
-    "message": "View at $1"
+    "message": "View $1 at $2"
```
[REQ] When we add variables to messages our current best practices are to include a description field that explains what those variables will be. There are examples in the messages.json file. I also realize that the original didn't have this but its a good time to add it now that we are extending it

brad-decker commented on ui/pages/swaps/awaiting-swap/view-on-ether-scan-link/view-on-ether-scan-link.js:
```diff
@@ -12,6 +12,8 @@ export default function ViewOnEtherScanLink({
 }) {
   const t = useContext(I18nContext);
 
+  const blockExplorerViewAction = 'Swap';
```
[NIT/Weakly held opinion/change my mind] while extracting constants like this is generally good, the likelihood of the object changing in the context of its location is slim to none without major refactors. I have an ever-so-slight preference of seeing the literal string value in the `t()` callsite because it is less contextual overhead to read and understand the output.

darkwing replied:
Could we should localize all of the `blockExplorerViewAction` values?  If they're front-facing to the user, I could see that being beneficial

brad-decker reviewed (changes requested):
This will make a lot of people very happy. 🎉 only minor feedback.

brad-decker commented:
Looks like you'll need to update some unit tests that look for the `view on etherscan` string

darkwing reviewed (changes requested):
A few nits from Brad and myself.

segun committed:
Fix Swap tests checking for View $1 at or View $1 on

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

segun committed:
Linter Fixes

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

segun committed:
Fix  ui/hooks/useTransactionDisplayData.test.js expected result should
be May 13, 2020

Update Jest snapshot for view on etherscan test

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

segun committed:
Add description of the variables in messages.

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

segun committed:
1. localize all of the blockExplorerViewAction values
2. Apply this nit. https://github.com/MetaMask/metamask-extension/pull/12100#discussion_r708343532

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

segun committed:
Lint fixes.

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

segun committed:
Fix locale value not used on GUI error lint error.

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

segun committed:
Reverted this commit, on circle ci, the date is May 12.

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

brad-decker reviewed (dismissed):
LGTM great work

adonesky1 commented:
Looks like this needs a rebase @segun!

Gudahtt commented:
It looks like this branch was messed up during a rebase; this includes _way_ more commits than it should.

mcmire reviewed (dismissed):
Makes sense to me.

Gudahtt commented:
>Looks like descriptions need to be added to all of the other languages :(

There is a script to help with this: `yarn verify-locales:fix`

Gudahtt reviewed (changes requested):
Because you're adding a new field to the localized message, this will now break for non-English locales that don't accept the second parameter.  The equivalent messages for other locales will need to be removed and replaced with new translations.

Gudahtt commented on app/_locales/en/messages.json:
```diff
@@ -259,6 +259,18 @@
     "message": "Welcome to MetaMask Beta"
   },
+  "blockExplorerAccountAction": {
+    "message": "Account"
```
Ideally these would have descriptions that link them back up to the `viewOnCustomBlockExplorer` and `viewOnEtherscan` messages, so that translators know which context these words will be used in.

segun committed:
Add description to link block explore actions to where it's used.

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

Gudahtt commented on app/_locales/am/messages.json:
```diff
@@ -1125,13 +1125,16 @@
   },
   "viewOnCustomBlockExplorer": {
-    "message": "በ $1ይመልከቱ"
+    "message": "በ $1ይመልከቱ",
```
To clarify, what I meant is that these three messages should be removed from non-English locales completely. Right now the wallet will throw an error when trying to display these messages because they're expected to have a `$1` in them, but there is no `$1`.

The easiest way to delete them all is to rename the English messages, run `yarn verify-locales:fix` (which will delete them in each other locale), then rename the English ones back to the original names.

segun replied:
I see, I ran :fix without renaming first.

segun committed:
Fix missing messages.

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

Gudahtt reviewed (approved):
LGTM!

mcmire reviewed (approved):
Nice, looks good!
